### PR TITLE
placement, kv: support RuleGroup configuration

### PR DESCRIPTION
**Pull request by disksing**
### What problem does this PR solve?
Support RuleGroup level Index and Override properties.
Tracking issue: https://github.com/pingcap/pd/issues/2721

### What is changed and how it works?
- kv storage L/S methods
- update ruleManager
- add unit test

### Check List

Tests
- Unit test

Code changes
- Has persistent data change

### Release note
- No release note

### CODE REVIEW

File: server/core/storage.go [hunk 1]
```diff
@@ -237,15 +234,44 @@ func (s *Storage) DeleteRule(ruleKey string) error {
...
+}
+
+// SaveRuleGroup stores a rule group config to storage.
+func (s *Storage) SaveRuleGroup(groupID string, group interface{}) error {
+	return s.SaveJSON(ruleGroupPath, groupID, group)
+}
+
+// DeleteRuleGroup removes a rule group from storage.
+func (s *Storage) DeleteRuleGroup(groupID string) error {
+	return s.Base.Remove(path.Join(ruleGroupPath, groupID))
```
Comment (rleungx) [inline server/core/storage.go:247]: How about directly using `s.Remove`?

File: server/schedule/placement/rule_manager.go [hunk 2]
```diff
@@ -125,6 +130,21 @@ func (m *RuleManager) loadRules() error {
...
+func (m *RuleManager) loadGroups() error {
+	err := m.store.LoadRuleGroups(func(k, v string) {
+		var g RuleGroup
+		if err := json.Unmarshal([]byte(v), &g); err != nil {
+			log.Error("failed to unmarshal rule group", zap.String("group-id", k), zap.Error(errs.ErrLoadRuleGroup.FastGenByArgs()), zap.NamedError("cause", err))
+			return
+		}
+		m.groups[g.ID] = &g
+	})
+	if err != nil {
```
Comment (rleungx) [inline server/schedule/placement/rule_manager.go:142]: How about `return err`?

Comment (disksing) [reply server/schedule/placement/rule_manager.go:142]: good idea

**Pull request merged**